pyccel/codegen/printing/ccode.py

```diff
@@ -1231,11 +1234,12 @@ def _print_NumpyUfuncBase(self, expr):
             numpy.cos(x) ==> cos(x)
 
         """
-        # add necessary include
         self.add_import(c_imports['math'])
         type_name = type(expr).__name__
         try:
             func_name = numpy_ufunc_to_c_float[type_name]
+            if func_name in ('sign'):
+                self.add_import(c_imports['numpy_c'])
```

Comment:
Are you sure that `NumpySign` is general enough to not need its own `_print_NumpySign` function? I'm not sure that the types are handled correctly here. `np.sign(3)` returns a `numpy.int64` but here you will have a double. This will cause problems if you translate:

```
print(np.sign(3))
```

Reply:
When translating to fortran I obtain:
```
At line 9 of file /home/emily/Code/test/tmp/__pyccel__/prog_tmp2.f90 (unit = 6, file = 'stdout')
Fortran runtime error: Expected INTEGER for item 1 in formatted transfer, got REAL
(I0)
 ^

Error termination. Backtrace:
#0  0x7f9693748700 in ???
#1  0x7f9693749259 in ???
#2  0x7f9693749f3f in ???
#3  0x7f969396fcc1 in ???
#4  0x7f969397c9d2 in ???
#5  0x7f9693980085 in ???
#6  0x7f96939807e3 in ???
#7  0x562508f6e341 in ???
#8  0x562508f6e0ae in ???
#9  0x7f9693559082 in __libc_start_main
	at ../csu/libc-start.c:308
#10  0x562508f6e0ed in ???
#11  0xffffffffffffffff in ???
```
When translating to C I obtain a compilation warning:
```
/home/emily/Code/pyccel/pyccel/codegen/compiling/compilers.py:431: UserWarning: /home/emily/Code/test/tmp/__pyccel__/prog_tmp2.c: In function ‘main’:
/home/emily/Code/test/tmp/__pyccel__/prog_tmp2.c:8:15: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 2 has type ‘double’ [-Wformat=]
    8 |     printf("%ld\n", sign(3.0));
      |             ~~^     ~~~~~~~~~
      |               |     |
      |               |     double
      |               long int
      |             %f

  warnings.warn(UserWarning(err))
```
then:
```
140725433417880
```

Reply:
I think the reason you get that value is because of the format string, but the actual result was fine.

Reply:
You are correct. The format string matches the type that pyccel tells it to expect (the dtype/precision of `NumpySign`), but your function does not return the type that it says it does. Hence the wrong printing. This should be handled by your function **not** by `_print_PythonPrint`.

Reply:
https://github.com/pyccel/pyccel/pull/1169/commits/d4745f9dc505e1f41dfad8e5055377eb82e30489 https://github.com/pyccel/pyccel/pull/1169/commits/da2619bf2fbda70678123e49267e344d601002d8 This issue is fixed new

Reply:
Could you add some tests in the `tests/pyccel/` folder with a few different types to confirm this please